app/scripts/models/profile-image.js

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This model abstracts profile images
+
+'use strict';
+
+define([
+  'backbone',
+  'lib/promise',
+  'lib/image-loader',
+  'lib/profile-errors'
+], function (Backbone, p, ImageLoader, ProfileErrors) {
+  var ProfileImage = Backbone.Model.extend({
+    defaults: {
+      url: undefined,
+      id: undefined,
+      img: undefined
+    },
+
+    fetch: function () {
+      var self = this;
+      if (! self.has('url')) {
+        return p();
+      }
+      return ImageLoader.load(this.get('url'))
+        .then(function (img) {
+          self.set('img', img);
+        }, function () {
+          return p.reject(ProfileErrors.toError('IMAGE_LOAD_ERROR'));
```

Comment:
Should the original error returned by the ImageLoader be saved to the returned error so that it can be logged? Is this error logged eventually?